private/pkg/manifest/manifest.go

```diff
@@ -93,6 +110,19 @@ func (d *Digest) Hex() string {
 	return d.hexstr
 }
 
+// Valid checks if the current digest matches a passed content, and any error
+// attempting to validate.
+func (d *Digest) Valid(content io.Reader) (bool, error) {
```

Comment:
I'm not sure I like this style. It forces Digest to know how to construct hashers; in this case it always constructs a new one.

Instead Digests can have equality defined:

```go
func (d *Digest) Equal(other *Digest) bool {
    return d.dtype == other.dtype && bytes.Equal(d.digest, other.digest)
}
```

Then you can have a Shake digest constructor, akin to your `shake256DigestFrom`, that can return Digests for content. Validation becomes this:

```go
digest.Equal(newShake256Digester().Sum(content))
```

Reply:
I like that idea, but then I'm thinking we should take it one step further: have a `type DigestType string`, have a `type Digester interface{ Digest(content io.Reader) (*Digest, error) }`, export available digest types (for now just shake256), and make all Manifest/Digest constructors receive the type.

Reply:
Done, created a new `Digester` interface for this equality logic, much cleaner, thanks!